src/bun.js/bindings/webcore/Worker.cpp

```diff
@@ -261,6 +262,12 @@ ExceptionOr<void> Worker::postMessage(JSC::JSGlobalObject& state, JSC::JSValue m
 
 void Worker::terminate()
 {
+
+    // Don't try to terminate an already terminated worker
+    if (m_terminationFlags & TerminatedFlag) {
```

Comment:
This isn't threadsafe. Fortunately, the `fetch_or` call returns the previous value, so you can do the check there. ANd since it's an or, it won't unset the terminated flag.